File: src/main/java/com/zhongan/devpilot/completions/inline/CompletionAdjustment.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by FernFlower decompiler)
//

package com.zhongan.devpilot.completions.inline;

import com.zhongan.devpilot.completions.general.SuggestionTrigger;
import com.zhongan.devpilot.completions.requests.AutocompleteRequest;
import com.zhongan.devpilot.completions.requests.AutocompleteResponse;

public abstract class CompletionAdjustment {
    private boolean cachedOnly = false;

    public CompletionAdjustment() {
    }

    public abstract SuggestionTrigger getSuggestionTrigger();

    public CompletionAdjustment withCachedOnly() {
        this.cachedOnly = true;
        return this;
    }

    public AutocompleteRequest adjustRequest(AutocompleteRequest request) {
        request.setCachedOnly(this.cachedOnly);
        return this.adjustRequestInner(request);
    }

    protected AutocompleteRequest adjustRequestInner(AutocompleteRequest autocompleteRequest) {
        return autocompleteRequest;
    }

    public AutocompleteResponse adjustResponse(AutocompleteResponse autocompleteResponse) {
        return autocompleteResponse;
    }
}

File: src/main/java/com/zhongan/devpilot/completions/inline/CompletionAdjustment.kt
package com.zhongan.devpilot.completions.inline

import com.zhongan.devpilot.completions.requests.AutocompleteRequest
import com.zhongan.devpilot.completions.requests.AutocompleteResponse
import com.zhongan.devpilot.completions.general.SuggestionTrigger

abstract class CompletionAdjustment {
    abstract val suggestionTrigger: SuggestionTrigger
    var cachedOnly: Boolean = false

    fun withCachedOnly(): CompletionAdjustment {
        cachedOnly = true
        return this
    }

    fun adjustRequest(request: AutocompleteRequest): AutocompleteRequest {
        request.cached_only = cachedOnly
        return adjustRequestInner(request)
    }

    protected open fun adjustRequestInner(autocompleteRequest: AutocompleteRequest): AutocompleteRequest = autocompleteRequest
    open fun adjustResponse(autocompleteResponse: AutocompleteResponse): AutocompleteResponse = autocompleteResponse
}
